### ISSUE TITLE
Option to disable access to GitHub and its prompt

### ISSUE THREAD

**Issue by heavywatal** [cschleiden/vscode-github-actions#167]
Some users (including me) may want to use this extension just for yaml editing, not for monitoring workflow status. It would be nice if frequent prompt and access to GitHub could be disabled. Currently, the prompt appears every time a project with GitHub origin is opened and every time an action yaml file tab gets focused.

**Comment by cschleiden** [comment]
Interesting 🤔 I hadn't considered this need. Some of the value comes from the connection to GitHub, like the validation of `with` parameters when using third-party actions, secret names, and runner labels. That being said, it might be possible to make that optional, or at least to prompt only once.

**Comment by sparr** [comment]
I, too, would like this feature. I cannot give this extension full access to my account and organizations.

**Comment by sparr** [comment]
> As mentioned above, the only thing that could be done is to provide only the static YAML editing features without displaying workflows and runs and dynamic auto-completion as suggested in #167

I think the language server could still provide more than just static editing. Being able to autocomplete the ids of steps and jobs in the current workflow or other workflows in the same repo, which can come from files like any other language server does it, would be very helpful.

**Comment by ayoh1** [comment]
+1 here as I wanted this for yaml validation and github login would require work org to approve the app access and I want to avoid that as its not needed... Just adding another use case to the issue

**Comment by ronjouch** [comment]
> Currently, the prompt appears every time a project with GitHub origin is opened and every time an action yaml file tab gets focused

A precision: the prompt appears more frequently than this. With a yml file open, it opens all the time, repeatedly, every time I hit a key or click on a piece of text. Makes the extension unusable.

**Comment by cschleiden** [comment]
We have updated the prompt and sign-in flow. Please try again if that helps with what you've been experiencing.

**Comment by sparr** [comment]
Thank you very much. Dismissing the prompt seems to work now, then syntax highlighting and language service functions are operational. I'll search and open another issue for a remaining lesser concern.

**Comment by heavywatal** [comment]
Thank you, @cschleiden, it works great. Aside from the one sparr reported in 224, a minor problem I had was to find a way to revoke authentication and to get the prompt again after revoking. My solution:

to revoke authentication
: Go to GitHub Settings > Integrations / Applications > Authorized OAuth Apps

to get prompted again to authenticate
: Try some random GitHub operations on VSCode. (Sorry I don't remember which exactly was the trigger)
